crates/kubelet/src/kubelet.rs

```diff
@@ -49,6 +55,97 @@ pub enum Phase {
 pub struct Status {
     pub phase: Phase,
     pub message: Option<String>,
+    pub container_statuses: Vec<ContainerStatus>,
+}
+
+/// A struct that contains all necessary information for a waiting status, meant
+/// to be used with the [ContainerStatus] enum
+#[derive(Clone, Debug)]
+pub struct WaitingStatus {
+    /// The timestamp of when this status was reported
+    pub timestamp: DateTime<Utc>,
+    /// A human readable string describing the why it is in a waiting status
+    pub message: String,
+}
+
+/// A struct that contains all necessary information for a running status, meant
+/// to be used with the [ContainerStatus] enum
+#[derive(Clone, Debug)]
+pub struct RunningStatus {
+    /// The timestamp of when this status was reported
+    pub timestamp: DateTime<Utc>,
+}
+
+/// A struct that contains all necessary information for a terminated status,
+/// meant to be used with the [ContainerStatus] enum
+#[derive(Clone, Debug)]
+pub struct TerminatedStatus {
+    /// The timestamp of when this status was reported
+    pub timestamp: DateTime<Utc>,
+    /// A human readable string describing the why it is in a terminating status
+    pub message: String,
+    /// Should be set to true if the process exited with an error
+    pub failed: bool,
+}
+
+/// ContainerStatus is a simplified version of the Kubernetes container status
+/// for use in providers. It allows for simple creation of the current status of
+/// a "container" (a running wasm process) without worrying about a bunch of
+/// Options. Use the [ContainerStatus::to_kubernetes] method for converting it
+/// to a Kubernetes API status
+#[derive(Clone, Debug)]
+pub enum ContainerStatus {
+    Waiting(WaitingStatus),
```

Comment:
Do the associated data of each enum variant need to be separate structs? Would it be simpler to do:

```rust
pub enum ContainerStatus {
    Waiting {
        /// The timestamp of when this status was reported
        pub timestamp: DateTime<Utc>,
        /// A human readable string describing the why it is in a waiting status
        pub message: String,
     },
    ....
```

Reply:
I honestly forgot this was an option. Will switch out